hyperspy/signal.py

```diff
@@ -3509,6 +3516,17 @@ def __deepcopy__(self, memo):
         for oaxis, caxis in zip(self.axes_manager._axes,
                                 dc.axes_manager._axes):
             caxis.navigate = oaxis.navigate
+
+        if dc.metadata.has_item('Markers'):
+            temp_marker_dict = dc.metadata.Markers.as_dictionary()
+            markers_dict = {}
+            for marker_name in temp_marker_dict.keys():
+                marker = dict2marker(
+                        temp_marker_dict[marker_name], marker_name)
+                if marker is not False:
+                    marker.axes_manager = dc.axes_manager
+                    markers_dict[marker_name] = marker
+            dc.metadata.Markers = markers_dict
```

Comment:
I made a new function in drawing/markers.py: https://github.com/hyperspy/hyperspy/pull/1431/files#diff-25b050d21071c0d2c1d0b5524030835aR221